ubtue_SAGE_Journals.js

```diff
@@ -0,0 +1,392 @@
+
+function detectWeb(doc, url) {
+	if (url.includes('/abs/10.') || url.includes('/full/10.') || url.includes('/pdf/10.') || url.includes('/doi/10.')) {
+		return "journalArticle";
+	}
+	else if (getSearchResults(doc, true)) {
+		return "multiple";
+	}
+	return false;
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '(//div|//span)[contains(@class, "art_title")]/a[contains(@href, "/doi/full/10.") or contains(@href, "/doi/abs/10.") or contains(@href, "/doi/pdf/10.")][1]');
+	for (var i = 0; i < rows.length; i++) {
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		href = href.replace("/doi/pdf/", "/doi/abs/");
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+
+function getPublishedPrintYearFromJSON(json) {
+	if ("published-print" in json) {
+		if ("date-parts" in json["published-print"]) {
+			return json["published-print"]["date-parts"][0][0];
+		}
+	}
+	return undefined;
+}
+
+
+function postProcess(doc, item) {
+	// remove partial DOIs stored in the pages field of online-first articles
+	if (item.DOI) {
+		var doiMatches = item.DOI.match(/\b(10[.][0-9]{4,}(?:[.][0-9]+)*\/((?:(?!["&'<>])\S)+))\b/);
+		if (doiMatches) {
+			var secondPart = doiMatches[2];
+			if (item.pages === secondPart) item.pages = "";
+		}
+	}
+}
+
+
+function scrapedAdditions(item, doc, doi) {
+	var abstract = ZU.xpathText(doc, '//article//div[contains(@class, "abstractSection")]/p');
+	if (abstract) {
+        item.abstractNote = abstract;
+	}
+
+	// ubtue: also add translated abstracts
+	var ubtueabstract = ZU.xpathText(doc, '//article//div[contains(@class, "tabs-translated-abstract")]/p | //*[contains(concat( " ", @class, " " ), concat( " ", "abstractInFull", " " ))]');
+	if (ubtueabstract) {
+        item.abstractNote += "\n\n" + ubtueabstract;
+	}
+
+	var tagentry = ZU.xpathText(doc, '//kwd-group[1] | //*[contains(concat( " ", @class, " " ), concat( " ", "hlFld-KeywordText", " " ))]');
+	if (tags) {
+		item.tags = tagentry.split(",");
+	}
+	// ubtue: add tags "Book Review" if "Review Article"
+	var articleType = ZU.xpath(doc, '//span[@class="ArticleType"]/span');
+	if (articleType) {
+		for (let r of articleType) {
+			let reviewDOIlink = r.innerHTML;
+			if (reviewDOIlink.match(/Review Article/)) {
```

Comment:
Maybe Change `/Review Article/` to `/Review\s+Article/` to detect non-breakable spaces and other unicode blanks

Reply:
(multiple occurences in this translator)